specifyweb/frontend/js_src/lib/components/DataModel/businessRuleDefs.ts

```diff
@@ -216,46 +217,50 @@ export const businessRuleDefs: MappedBusinessRuleDefs = {
             if (
               collectionObject !== undefined &&
               collectionObject.specifyTable.name === 'CollectionObject'
-            )
-              void collectionObject
-                .rgetPromise('collectionObjectType', true)
-                .then(
-                  async (
-                    coType: SpecifyResource<CollectionObjectType> | null
-                  ) => {
-                    const resolvedCoType =
-                      coType === null
-                        ? await collectionObject
-                            .rgetPromise('collection', true)
-                            .then(async (collection) =>
-                              collection.rgetPromise('collectionObjectType')
-                            )
-                        : coType;
-                    /*
-                     * Have to set save blockers directly here to get this working.
-                     * Since following code has to wait for above rgetPromise to resolve, returning a Promise<BusinessRuleResult> for validation here is too slow and
-                     * does not get captured by business rules.
-                     */
-                    if (
-                      resolvedCoType?.get('taxonTreeDef') ===
-                      (taxon?.get('definition') ?? '')
-                    ) {
-                      setSaveBlockers(
-                        determination as SpecifyResource<AnySchema>,
-                        determination.specifyTable.field.taxon,
-                        [],
-                        DETERMINATION_TAXON_KEY
-                      );
-                    } else {
-                      setSaveBlockers(
-                        determination as SpecifyResource<AnySchema>,
-                        determination.specifyTable.field.taxon,
-                        [formsText.invalidTree()],
-                        DETERMINATION_TAXON_KEY
-                      );
-                    }
+            ) {
+              void f
+                .all({
+                  defaultType:
+                    collectionObject.get('collectionObjectType') === undefined
+                      ? collectionObject
+                          .rgetPromise('collection')
+                          .then(async (collection) =>
+                            collection.rgetPromise('collectionObjectType')
+                          )
+                          .then((coType) => coType ?? undefined)
+                      : undefined,
```

Comment:
Maybe we need a `customInit` as well for CoType which sets the default cotype in a CO form? Could extract this logic into a function and use it there. What do you think?

Reply:
we can create a new issue to set default cotype :+1: